### PR TITLE
Copilot/update type hints in files again

### PR DESCRIPTION
**Pull request by aaishwarymishra**
This pull request modernizes type annotations throughout several metric modules in the codebase, replacing legacy `typing` constructs (like `Optional`, `Union`, `List`, and `Tuple`) with Python 3.9+ syntax (e.g., `|`, `list`, `tuple`). These changes improve code readability and maintainability, and ensure consistency across the codebase. No functional logic is changed—these are strictly type annotation updates.

**Type annotation modernization:**

* Replaced `Optional`, `Union`, `List`, and `Tuple` with Python 3.9+ native type hints (e.g., `str | None`, `list[torch.Tensor]`, `tuple[int, ...]`) in `ignite/metrics/classification_report.py`, `ignite/metrics/confusion_matrix.py`, `ignite/metrics/epoch_metric.py`, and `ignite/metrics/mean_average_precision.py`. [[1]](diffhunk://#diff-5bff0a9393c74577bc225499798c6e6194882fe34cd5c110e54bc0f03d61a66fL2-R2) [[2]](diffhunk://#diff-5bff0a9393c74577bc225499798c6e6194882fe34cd5c110e54bc0f03d61a66fL18-R20) [[3]](diffhunk://#diff-5bff0a9393c74577bc225499798c6e6194882fe34cd5c110e54bc0f03d61a66fL130-R130) [[4]](diffhunk://#diff-e73c2e9b856afa4e0c94f37145d50822e2a7515b50dc81a6ebf34895112e3998L2-R2) [[5]](diffhunk://#diff-e73c2e9b856afa4e0c94f37145d50822e2a7515b50dc81a6ebf34895112e3998L114-R116) [[6]](diffhunk://#diff-e73c2e9b856afa4e0c94f37145d50822e2a7515b50dc81a6ebf34895112e3998L155-R155) [[7]](diffhunk://#diff-e73c2e9b856afa4e0c94f37145d50822e2a7515b50dc81a6ebf34895112e3998L205-R205) [[8]](diffhunk://#diff-e73c2e9b856afa4e0c94f37145d50822e2a7515b50dc81a6ebf34895112e3998L274-R274) [[9]](diffhunk://#diff-e73c2e9b856afa4e0c94f37145d50822e2a7515b50dc81a6ebf34895112e3998L368-R368) [[10]](diffhunk://#diff-e73c2e9b856afa4e0c94f37145d50822e2a7515b50dc81a6ebf34895112e3998L433-R433) [[11]](diffhunk://#diff-bc59be9b209a7f952cfc6c77e2938ba4f94854b2ad32a9abe3985dd3a7ae69cbL2-R2) [[12]](diffhunk://#diff-bc59be9b209a7f952cfc6c77e2938ba4f94854b2ad32a9abe3985dd3a7ae69cbL80-R80) [[13]](diffhunk://#diff-bc59be9b209a7f952cfc6c77e2938ba4f94854b2ad32a9abe3985dd3a7ae69cbL93-R105) [[14]](diffhunk://#diff-bc59be9b209a7f952cfc6c77e2938ba4f94854b2ad32a9abe3985dd3a7ae69cbL120-R120) [[15]](diffhunk://#diff-47e7eabac627499ef580d80397243b6c00dcec3077257187a31a52fd2b671567L2-R2) [[16]](diffhunk://#diff-47e7eabac627499ef580d80397243b6c00dcec3077257187a31a52fd2b671567L17-R18) [[17]](diffhunk://#diff-47e7eabac627499ef580d80397243b6c00dcec3077257187a31a52fd2b671567L59-R67) [[18]](diffhunk://#diff-47e7eabac627499ef580d80397243b6c00dcec3077257187a31a52fd2b671567L111-R114) [[19]](diffhunk://#diff-47e7eabac627499ef580d80397243b6c00dcec3077257187a31a52fd2b671567L142-R151) [[20]](diffhunk://#diff-47e7eabac627499ef580d80397243b6c00dcec3077257187a31a52fd2b671567L300-R300) [[21]](diffhunk://#diff-47e7eabac627499ef580d80397243b6c00dcec3077257187a31a52fd2b671567L318-R318) [[22]](diffhunk://#diff-47e7eabac627499ef580d80397243b6c00dcec3077257187a31a52fd2b671567L360-R360) [[23]](diffhunk://#diff-47e7eabac627499ef580d80397243b6c00dcec3077257187a31a52fd2b671567L370-R370)

**Consistency improvements:**

* Ensured consistent use of updated type annotations for parameters, return types, and internal variables across all affected metric classes and utility functions. [[1]](diffhunk://#diff-5bff0a9393c74577bc225499798c6e6194882fe34cd5c110e54bc0f03d61a66fL18-R20) [[2]](diffhunk://#diff-e73c2e9b856afa4e0c94f37145d50822e2a7515b50dc81a6ebf34895112e3998L114-R116) [[3]](diffhunk://#diff-bc59be9b209a7f952cfc6c77e2938ba4f94854b2ad32a9abe3985dd3a7ae69cbL80-R80) [[4]](diffhunk://#diff-47e7eabac627499ef580d80397243b6c00dcec3077257187a31a52fd2b671567L17-R18) [[5]](diffhunk://#diff-47e7eabac627499ef580d80397243b6c00dcec3077257187a31a52fd2b671567L142-R151)

**No logic changes:**

* All changes are strictly related to type annotations; no runtime or functional behavior has been affected. (All references)

### CODE REVIEW

File: ignite/metrics/classification_report.py [hunk 1]
```diff
@@ -1,5 +1,5 @@
 import json
-from typing import Callable, Collection, Dict, List, Literal, Optional, Union
+from typing import Callable, Collection, Literal
```
Comment (TahaZahid05) [inline ignite/metrics/classification_report.py]: Move `Callable` and `Collection` to `collections.abc`
```suggestion
from collections.abc import Callable, Collection
from typing import Literal
```

File: ignite/metrics/confusion_matrix.py [hunk 2]
```diff
@@ -1,5 +1,5 @@
 import numbers
-from typing import Callable, Optional, Sequence, Tuple, Union
+from typing import Callable, Sequence
```
Comment (TahaZahid05) [inline ignite/metrics/confusion_matrix.py]: Move `Callable` and `Sequence` to `collections.abc`

```suggestion
from collections.abc import Callable, Sequence
```

File: ignite/metrics/epoch_metric.py [hunk 3]
```diff
@@ -1,5 +1,5 @@
 import warnings
-from typing import Callable, cast, List, Optional, Tuple, Union
+from typing import Callable, cast
```
Comment (TahaZahid05) [inline ignite/metrics/epoch_metric.py]: Move `Callable` to `collections.abc`

```suggestion
from collections.abc import Callable
from typing import cast
```

File: ignite/metrics/mean_average_precision.py [hunk 4]
```diff
@@ -1,5 +1,5 @@
 import warnings
-from typing import Callable, List, Optional, Sequence, Tuple, Union, cast
+from typing import Callable, Sequence, cast
 
 import torch
 from typing_extensions import Literal
```
Comment (TahaZahid05) [inline ignite/metrics/mean_average_precision.py]: Import `Literal` from `typing` instead
```suggestion
```

File: ignite/metrics/mean_average_precision.py [hunk 5]
```diff
@@ -1,5 +1,5 @@
 import warnings
-from typing import Callable, List, Optional, Sequence, Tuple, Union, cast
+from typing import Callable, Sequence, cast
```
Comment (TahaZahid05) [inline ignite/metrics/mean_average_precision.py]: Move `Callable` and `Sequence` to `collections.abc`

```suggestion
from collections.abc import Callable, Sequence
from typing import cast, Literal
```

**Comment by aaishwarymishra** [comment]
why these few tests getting timed out?

**Comment by TahaZahid05** [comment]
@aaishwarymishra I think there are some issues in the `pytorch-nightly` test itself. If the rest are passing, I guess it is fine.

**Review by vfdev-5 — APPROVED** [review]
